### PR TITLE
Remove dot-files not needed and update .gitignore

### PR DESCRIPTION
**Pull request by dsmiley**
These dot-files seem like not worth keeping.

One of them was referenced in `.gitignore` so that brought my attention to remove some things there as well.  Maybe could be another PR; I can split if asked.  If someone uses less common tools (e.g. "syncthinghing"), they can add ignore rules for their own checkout, or put them in a common file (like in your home dir) and refer to it via adding a `core.excludesFile` git config reference.  That could even go in `~/.gitconfig`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,8 +11,6 @@ build/
 **/*.ipr
 **/*.iws
 out/
-#    IntelliJ creates this folder, ignore.
-/dev-tools/missing-doclet/out/
```
Comment (dsmiley) [inline .gitignore:15]: obsoleted by the `out/` added a year later on the line above

File: .gitignore [hunk 2]
```diff
@@ -24,22 +22,20 @@ out/
 
 .editorconfig
 
-# syncthing
-.stignore
```
Comment (dsmiley) [inline .gitignore:28]: @markrmiller  added this one once

File: .inferconfig
Comment (dsmiley) [inline .inferconfig:1]: Was an interesting tool used by Lift, but Lift is gone

**Review by mkhludnev — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (janhoy) [reply .gitignore:28]: I see no harm in this, as it reduces the chance of any future contributor accidentally committing this file if they use such tool. A few extra lines in gitignore is tolerable. We’re not adding every possible tool in here, but devs will add such as they experience the pain with a certain tool.

**Comment by dsmiley** [comment]
Will merge in 24 hours or so

**Pull request merged**